eclipse/che
# feat: migrating from TSLint to ESLint and formatting with Prettier added pre-commit hook with Husky

### What does this PR do?

- Migrated to eslint from tslint
- Added formatter prettier
- Added pre-commit hook to apply linter check and auto format with Husky
- All fixups on rules applied 

### Screenshot/screencast of this PR

NA

### What issues does this PR fix or reference?

#22387 

### How to test this PR?

Make some changes and commit. Try to use double quotes instead single. New formatter should fix it.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
